lightning/src/ln/invoice_utils.rs

```diff
@@ -808,6 +804,123 @@ mod test {
 		assert_eq!(events.len(), 2);
 	}
 
+	#[test]
+	fn payment_metadata_end_to_end_for_invoice_with_amount() {
```

Comment:
@TheBlueMatt Not sure where you prefer these tests go. I guess we can move them once / if `invoice_utils.rs` is removed. I'm somewhat inclined to rename `bolt11_payment.rs` to `bolt11_payment_tests.rs`.

Reply:
Done! [pr3617.06](https://github.com/shaavan/rust-lightning/commits/pr3617.06)
I have renamed the file to bolt11_payment_tests.rs, and moved the new introduced function within it.
Thanks a lot for the suggestion! ✨